lib/galaxy/datatypes/binary.py

```diff
@@ -1403,10 +1403,7 @@ def sniff_table_names(self, filename, table_names):
             tables_query = "SELECT name FROM sqlite_master WHERE type='table' ORDER BY name"
             result = c.execute(tables_query).fetchall()
             result = [_[0] for _ in result]
-            for table_name in table_names:
-                if table_name not in result:
-                    return False
-            return True
+            return set(result) == set(table_names)
```

Comment:
You can  also do `result = (_[0] for _ in result)`. And here I'd use `set(result).issubset(set(table_names))` to maintain the logic.

Reply:
Also `set(result).issubset(table_names)` works.